internal/compiler/program.go

```diff
@@ -255,13 +264,55 @@ func equalCheckJSDirectives(d1 *ast.CheckJsDirective, d2 *ast.CheckJsDirective)
 	return d1 == nil && d2 == nil || d1 != nil && d2 != nil && d1.Enabled == d2.Enabled
 }
 
-func (p *Program) SourceFiles() []*ast.SourceFile { return p.files }
-func (p *Program) Options() *core.CompilerOptions { return p.opts.Config.CompilerOptions() }
-func (p *Program) Host() CompilerHost             { return p.opts.Host }
+func (p *Program) SourceFiles() []*ast.SourceFile            { return p.files }
+func (p *Program) Options() *core.CompilerOptions            { return p.opts.Config.CompilerOptions() }
+func (p *Program) CommandLine() *tsoptions.ParsedCommandLine { return p.opts.Config }
+func (p *Program) Host() CompilerHost                        { return p.opts.Host }
 func (p *Program) GetConfigFileParsingDiagnostics() []*ast.Diagnostic {
 	return slices.Clip(p.opts.Config.GetConfigFileParsingDiagnostics())
 }
 
+// ExtractUnresolvedImports returns the unresolved imports for this program.
+// The result is cached and computed only once.
+func (p *Program) ExtractUnresolvedImports() collections.Set[string] {
+	if p.cloned {
+		return *p.unresolvedImports
+	}
+	p.unresolvedImportsOnce.Do(func() {
+		p.unresolvedImports = p.extractUnresolvedImports()
+	})
+
+	return *p.unresolvedImports
```

Comment:
I still feel like this would make more sense as:

```suggestion
	p.unresolvedImportsOnce.Do(func() {
		if p.unresolvedImports == nil {
			p.unresolvedImports = p.extractUnresolvedImports()
		}
	})

	return *p.unresolvedImports
```

And no `cloned` field needed.

Reply:
(That and I'm not sure why this returns `collections.Set` instead of a pointer to it)

Reply:
Oh yeah, that would work